code/modules/antagonists/revenant/revenant.dm

```diff
@@ -60,18 +60,47 @@
 	var/essence_drained = 0 //How much essence the revenant will drain from the corpse it's feasting on.
 	var/draining = FALSE //If the revenant is draining someone.
 	var/list/drained_mobs = list() //Cannot harvest the same mob twice
+	var/list/thrall = list() //Preta only var, but it's used to keep in check who is the thrall
 	var/perfectsouls = 0 //How many perfect, regen-cap increasing souls the revenant has. //TODO, add objective for getting a perfect soul(s?)
 	var/generated_objectives_and_spells = FALSE
+	var/pickedghost = "Buggy Enigma. Please report this on github"
 
 /mob/living/simple_animal/revenant/Initialize(mapload)
 	. = ..()
 	AddSpell(new /obj/effect/proc_holder/spell/targeted/night_vision/revenant(null))
 	AddSpell(new /obj/effect/proc_holder/spell/targeted/revenant_transmit(null))
 	AddSpell(new /obj/effect/proc_holder/spell/aoe_turf/revenant/defile(null))
-	AddSpell(new /obj/effect/proc_holder/spell/aoe_turf/revenant/overload(null))
-	AddSpell(new /obj/effect/proc_holder/spell/aoe_turf/revenant/blight(null))
-	AddSpell(new /obj/effect/proc_holder/spell/aoe_turf/revenant/malfunction(null))
 	random_revenant_name()
+	if(type=="/mob/living/simple_animal/revenant") //so not an admin spawned specific revenant, lets randomly pick one
```

Comment:
No.

Reply:
.....

Reply:
yeah yeah there's a better way to do it

but like why not this one line check

Reply:
there shouldn't be any quotes for one, if someone changes the path and this is missed, then it'll silently fail since it's a string.